js/ViewThread.js

```diff
@@ -331,4 +334,62 @@ ViewThread.prototype.detectQuotedMessageId = function(text) {
 		quotedId = matches[1];
 	}
 	return quotedId;
-};
\ No newline at end of file
+};
+
+/**
+ * For every author currently displayed, fetch and display the avatar
+ */
+ViewThread.prototype.fetchAvatars = function() {
+	//console.log("récupération d'avatars !");
+	var lthis = this;
+	$('.thread-message').each(function() {
+		var avatar = null;
+		var messageId = $(this).attr('id').substr(4);
+		// get author email
+		var authorEmail = lthis.getAuthorEmailFromMsgId(messageId);
+		if (authorEmail != null) {
+			// is the avatar already in the cache ?
+			if (authorEmail in lthis.avatarCache) {
+				//console.log('trouvé dans le cache : ' + lthis.avatarCache[authorEmail]);
+				avatar = lthis.avatarCache[authorEmail];
+				// display it !
+				$(this).find('.author-image > img').attr('src', avatar);
+			} else {
+				// fetch it using the service
+				var url = lthis.config.avatarService.replace('{email}', authorEmail);
+				var that = this; // inception
```

Comment:
Instead of `that` or `lthis` and stuff, you could use real var name like `currentThreadMessage` or something